src/DFGNode.cpp

```diff
@@ -243,12 +284,67 @@ bool DFGNode::isLogic() {
   return false;
 }
 
-bool DFGNode::hasCombined() {
+bool DFGNode::isLut() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and op.compare("lut") == 0)
+    return true;
+  return false;
+}
+
+bool DFGNode::isDequantize() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and op.compare("Dequantize") == 0)
+    return true;
+  return false;
+}
+
+// Convert the fp8/fp16 to fp32 or int8/int16 to int32.
+bool DFGNode::isConvert() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and (op.compare("fpConvert") == 0 or op.compare("intConvert") == 0))
+    return true;
+  return false;
+}
+
+// Divison can also be a special operation.
+bool DFGNode::isDiv() {
+  if (m_opcodeName.compare("fdiv") == 0 or m_opcodeName.compare("div") == 0)
+    return true;
+  return false;
+}
+
+// Quantize the fp32 to fp8/fp16 or int32 to int8/int16.
+bool DFGNode::isQuantize() {
+  string op = getOpcodeName();
+  if (m_opcodeName.compare("call") == 0 and (op.compare("fpQuantize") == 0 or op.compare("intQuantize") == 0))
+    return true;
+  return false;
+}
+
+// used for specialized fusion (e.g. alu+mul and icmp+br can be regared as two kinds of complex nodes, so there are different tiles to support them)
+// type indicates the type of the combined node. (e.g. 0 for alu+mul, 1 for icmp+br)
+// type=-1 means the node is combined with any type, which is used for general fusion.
+bool DFGNode::hasCombined(int type) {
+  if (type < 0) return m_combined;
+  else return m_combinedtype[type];
+}
+
+// for getptr + load/store, we should not regard them as complex operations. 
+// in CGRA, getptr+l/s can be performed on the common tiles.
+// in DFG, it should be still regarded as complex nodes so that we will perform combining and tuning.
```

Comment:
Can you plz elaborate on this (`it should be still regarded as complex nodes so that we will perform combining and tuning`)?

Reply:
This function is useless and I deleted it.